lib/app/dispatcher.rb

```diff
@@ -93,9 +105,36 @@ def start_polling_jobs!
         Utility::ExceptionTracking.log_exception(e, 'connector service failed due to unexpected error.')
       end
 
-      def send_heartbeat(connector_id, service_type)
-        Utility::Logger.info("Sending a heartbeat on [#{connector_id} - #{service_type}]...")
-        Core::Heartbeat.send(connector_id, service_type)
+      def start_heartbeat_task!
+        Thread.new do
```

Comment:
What's the reasoning behind starting all of it in new threads? The work units are submitted to the thread pools anyway, right? Why make threads to create threads?

Reply:
Yes, the real work will be submitted to the pool, e.g. the heartbeat task, and the configuration. But we can't have a job polling loop, a heartbeat loop and a configuration loop running in one thread at the same time. That's why we need to have 3 threads to run the loop.

`Thread.new` is used here instead of thread pool because the threads here are not meant to be reused.